subsys/bluetooth/mesh/vnd/silvair_enocean_srv.c

```diff
@@ -335,6 +380,55 @@ const struct bt_mesh_model_op _bt_mesh_silvair_enocean_srv_op[] = {
 	BT_MESH_MODEL_OP_END,
 };
 
+static void delay_update(struct bt_mesh_model *model, uint32_t offset)
+{
+	uint32_t delay;
+
+	delay = delay_get(model->pub, BT_MESH_PUB_MSG_NUM(model->pub));
+	model->pub->msg->data[offset] = model_delay_encode(delay);
+}
+
+static int onoff_update_handler(struct bt_mesh_model *model)
+{
+	struct bt_mesh_onoff_cli *cli = model->user_data;
+
+	if (cli->pub_buf.len != 6 ||
+	    !bt_mesh_model_pub_is_retransmission(model)) {
+		return -EINVAL;
+	}
+
+	delay_update(model, 5);
+
+	return 0;
+}
+
+static int lvl_update_handler(struct bt_mesh_model *model)
+{
+	struct bt_mesh_lvl_cli *cli = model->user_data;
+	struct bt_mesh_silvair_enocean_button *button =
+		CONTAINER_OF(cli, struct bt_mesh_silvair_enocean_button, lvl);
+
+	if (cli->pub_buf.len != 9 ||
+	    !bt_mesh_model_pub_is_retransmission(model)) {
+		return -EINVAL;
+	}
+
+	if (button->state != BT_MESH_SILVAIR_ENOCEAN_STATE_DELTA) {
+		return 0;
+	}
+
+	delay_update(model, 8);
+
+	/** Check if this is the last publication in Phase B. */
```

Comment:
No, because in Phase D, the state changes to Idle and the execution will be stopped at line 417:
https://github.com/nrfconnect/sdk-nrf/blob/fa55aecf35b4defac1ae04bb178e9ed981e42bff/subsys/bluetooth/mesh/vnd/silvair_enocean_srv.c#L416-L418